packages/webamp-modern-2/src/skin/SystemObject.ts

```diff
@@ -162,6 +162,253 @@ export default class SystemObject extends BaseObject {
+  }
+
+  /**
+  * getExtFamily()
+  * 
+  * This one still return the same as getDecoderName()
+  * 
+  * //Gets registered family for given extension (i.e. mp3 -> "Audio")
+  * 
+  * @ret The registered family for given extension.
+  * @param ext The extension in question.
+  */
+  getextfamily(ext: string): string {
+    return "Audio";
+  }
+
+  /**
+   * 
+   * @param playitem 
+   * @returns Above as getExtFamily(), needs more investigating
+   */
+  getdecodername(playitem: string): string {
+    return "Nullsoft MPEG Decoder v4.103";
+  }
+
+  /**
+  * downloadMedia()
+  * 
+  * Hookable. Downloads a file from url and saves it as destination_filename.
+  * If destinationPath is "" it will be saved in CD Ripping dir (=getDownloadPath()).
+  * 
+  * @param  url  the file to be downloaded
+  * @param  destinationPath  the path you want to store the file
+  * @param  wantAddToML  set true if you wnt to add the file to ML database
+  * @param  notifyDownloadsList  set true in order to list the download in <DownloadsLis/> Object
+  */
+  downloadmedia(url: string, destinationPath: string, wantAddToML: boolean, notifyDownloadsList: boolean): string {
+    //TODO
+  }
+
+  /**
+   * Deprecated, destination_filename is just a dummy parameter as well as progress_dialog_title - use download() instead
+   * Requires 5.5
+   * @param url 
+   * @param destination_filename 
+   * @param progress_dialog_title 
+   */
+  downloadurl(url: string, destination_filename: string, progress_dialog_title: string) {
+    
+  } 
+
+  /**
+   * 
+   * @param url 
+   * @param success 
+   * @param filename 
+   */
+  ondownloadfinished(url: string, success: boolean, filename: string): string {
+    //TODO
+  }
+
+  /**
+   * 
+   * @param string returns the CD Ripping Directory
+   */
+  getdownloadpath(): string {
+    return "C:\\CD Rips";
+  }
+  /**
+   * 
+   * @param new_path Sets the CD Ripping Directory
+   */
+  setdownloadpath(new_path: string): string {
+    return "C:\\Music\\CD Rips";
+  }
+
+  /**
+  * enqueueFile()
+  * 
+  * Enqueque the requested file. Path and filename are required
+  * with proper extension (example: playFile("c:\music\mp3\file.mp3"); ).
+  * It also works with URL's (example: playFile("http://myshoutcast.com:8000"); ).
+  * 
+  * @param  playitem  The path and filename to play.
+  */
+  enqueuefile(playitem: string): string {
+    return "http://myshoutcast.com:8000";
+  } //Requires 5.5
+
+  /**
+  * playFile()
+  * 
+  * Play the requested file. Path and filename are required
+  * with proper extension (example: playFile("c:\music\mp3\file.mp3"); ).
+  * It also works with URL's (example: playFile("http://myshoutcast.com:8000"); ).
+  * 
+  * @param  playitem  The path and filename to play.
+  */
+  playfile(playitem: string): string {
+  return "C:\\Program Files\\Winamp\\demo.mp3"
+  }
+
+  /**
+   * //Requires 5.51
+   * @param fullfilename 
+   */
+  getfilesize(fullfilename: string): string {
+    return "FIXME"
+  } 
```

Comment:
I think this is supposed to return a number (`int` in Maki)

```suggestion
  getfilesize(fullfilename: string): number {
    return 100;
  } 
```